src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -54,19 +54,34 @@ public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
         }
 
         /// <summary>
-        /// This is called to query the list of debug targets  
+        /// Called by StartupProjectRegistrar to determine whether this project should appear in the Startup list.
         /// </summary>
-        public override Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsAsync(DebugLaunchOptions launchOptions)
+        public async Task<bool> CanBeStartupProjectAsync(DebugLaunchOptions launchOptions)
         {
-            return QueryDebugTargetsInternalAsync(launchOptions, fromDebugLaunch: false);
+            try
+            {
+                ILaunchProfile activeProfile = await GetActiveProfileAsync();
+
+                // Now find the DebugTargets provider for this profile
+                IDebugProfileLaunchTargetsProvider launchProvider =
+                    GetLaunchTargetsProvider(activeProfile) ?? throw new Exception();
```

Comment:
If you can't get a launch profile, it throws an  exception (caught below) and the function returns true. I guess that is for compat as well?

Reply:
Yeah... sounds redundant to throw an exception and catch it right away. Let me change this